From 41838e3a6c027fcafae9d8f8a2c32c7be5d8331b Mon Sep 17 00:00:00 2001
From: Johannes Thoma <johannes@johannesthoma.com>
Date: Wed, 24 Mar 2021 18:12:57 +0100
Subject: [PATCH] drbd_nl: initialize variables.

For MS C compiler using an uninitialized pointer is an error.
---
 drbd/drbd_nl.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drbd/drbd_nl.c b/drbd/drbd_nl.c
index c4e4b0f..051ee0e 100644
--- a/drbd/drbd_nl.c
+++ b/drbd/drbd_nl.c
@@ -5591,7 +5591,7 @@ int drbd_adm_dump_devices(struct sk_buff *skb, struct netlink_callback *cb)
 	KIRQL rcu_flags;
 	struct nlattr *resource_filter;
 	struct drbd_resource *resource;
-	struct drbd_device *device;
+	struct drbd_device *device = NULL;
 	int minor, err, retcode;
 	struct drbd_genlmsghdr *dh;
 	struct device_info device_info;
@@ -5711,12 +5711,15 @@ static void connection_to_statistics(struct connection_statistics *s, struct drb
 
 enum { SINGLE_RESOURCE, ITERATE_RESOURCES };
 
+/* warning C4701: potentially uninitialized local variable 'minor' used */
+#pragma warning (disable: 4701)
+
 int drbd_adm_dump_connections(struct sk_buff *skb, struct netlink_callback *cb)
 {
 	KIRQL rcu_flags;
 	struct nlattr *resource_filter;
 	struct drbd_resource *resource = NULL, *next_resource;
-	struct drbd_connection *connection;
+	struct drbd_connection *connection = NULL;
 	int err = 0, retcode;
 	struct drbd_genlmsghdr *dh;
 	struct connection_info connection_info;
@@ -5915,7 +5918,7 @@ int drbd_adm_dump_peer_devices(struct sk_buff *skb, struct netlink_callback *cb)
 	KIRQL rcu_flags;
 	struct nlattr *resource_filter;
 	struct drbd_resource *resource;
-	struct drbd_device *device;
+	struct drbd_device *device = NULL;
 	struct drbd_peer_device *peer_device = NULL;
 	int minor, err, retcode;
 	struct drbd_genlmsghdr *dh;
-- 
2.17.0

